csrc/layernorm_quant_kernels.cu

```diff
@@ -169,20 +196,37 @@ fused_add_rms_norm_static_fp8_quant_kernel(
 
 }  // namespace vllm
 
-void rms_norm_static_fp8_quant(torch::Tensor& out,     // [..., hidden_size]
-                               torch::Tensor& input,   // [..., hidden_size]
-                               torch::Tensor& weight,  // [hidden_size]
+// --- shared: match unfused launch exactly ---
+static inline int ln_block_threads_unified(int hidden_size) {
+  int threads = (hidden_size >= 1024) ? 256
+                : (hidden_size >= 512)
+                    ? 512
+                    : std::min(1024, ((hidden_size + 31) / 32) * 32);
+  // warp-align and clamp to [128, 1024]
+  threads = std::min(1024, std::max(128, ((threads + 31) / 32) * 32));
+  return threads;
+}
+
```

Comment:
reuse code here to avoid duplicate code
And please also add comments why we need this function?